src/axom/inlet/Container.hpp

```diff
@@ -269,6 +269,21 @@ bool matchesKeyType(const VariantKey& key)
   return false;
 }
 
+/*!
+ *******************************************************************************
+ * \brief Updates the set of unexpected names to reflect an user-requested access
+ * 
+ * \param [in] accessed_name The path within the input file that will be accessed
+ * \param [inout] unexpected_names The set of input file paths that have not yet
+ * been requested by the user
+ * 
+ * \note To maintain consistency, this function should always be followed by an
+ * access to a Reader
```

Comment:
"To maintain consistency, this function should always be followed by an access to a Reader". This seems easy to mess up. Is there a pattern you can use to avoid this? For example, what about an Reader which wraps another reader and keeps track of the unexpected names? With that pattern, `m_unexpected_names` would become a field of the wrapper proxy.

Reply:
I had considered something like this but was concerned about the additional complexity that it introduces.  I agree that it's possibly error-prone but on the other hand it's an internal method that currently only gets used in four places (in a type-independent context).  Any thoughts @white238?

Reply:
I agree this is a little confusing/easy to mess up but it is only in a few places and those can be handled by adding unit tests specifically for this.  I'm not sure the extra complexity is worth it at this point since it is not user-facing and I can't think of anything else that would go into this class. If we do we should add this language-agnostic Reader wrapper (i hope it is named `ReaderKeeper`).